src/PhysicsEngine.cpp

```diff
@@ -185,3 +184,257 @@ void ForwardEuler::march_step(double tStamp, double tStep) {
+        double normal_coef = c_Na * alpha;
+        
+        // 1.225 is temporary density of air
+        double normal_force_mag = 0.5 * normal_coef * vel_rf.magnitude2() * area * 1.225;
+        normal_force_rf.x = (-vel_rf.x);
+        normal_force_rf.y = (-vel_rf.y);
+        normal_force_rf.z = 0;
+
+        normal_force_rf.normalize();
+        normal_force_rf = normal_force_rf * normal_force_mag;
+
+        double drag_mag = 0.5 * drag_coef * vel_rf.magnitude2() * area * 1.225;
+        Vector3 drag_rf(0, 0, -(drag_mag));
+
+        aero_force_rf = normal_force_rf + drag_rf;
+        aero_torque_rf = Cp_vect_rf.cross(aero_force_rf);
+    } else {
+        aero_torque_rf.x = 0;
+        aero_torque_rf.y = 0;
+        aero_torque_rf.z = 0;
+    }
+
+    Vector3 net_torque_if = rocket_.r2i(aero_torque_rf);
+
+    return net_torque_if;
+}
+
+/**
+ * @brief Calculates a possible rocket state based on the initial and inputed state
+ * 
+ * @param tStamp 
+ * @param tStep 
+ * @param k 
+ * @return State 
+ */
+State RungeKutta::calc_state(double tStamp, double tStep, State k) {
+    
+    // the rocket's initial state (k1), regardless of which state is being calculated
+    Vector3 vel_initial = rocket_.get_r_dot();
+    Vector3 ang_vel_initial = rocket_.get_w_vect();
+    Quaternion<double> orient_true = rocket_.get_q_ornt();
+
+    double inertia[9];
+    rocket_.get_I(inertia);
+
+    Quaternion<double> orient = calc_orient(tStep, k.ang_vel, orient_true);
+    rocket_.set_q_ornt(orient);  // sets the orientation to the current state in order to calculate net forces
+    
+    // Euler Step: y = x + (dx * t)
+    Vector3 vel_k = vel_initial + (k.accel * tStep); // Euler step combinging initial velocity with acceleration from the previous state
+    Vector3 accel_k = calc_net_force(tStamp, k.vel) / rocket_.get_mass();  // Calulate acceleration using forces from the previous state
+    Vector3 ang_vel_k = ang_vel_initial + (k.ang_accel * tStep);
+    Vector3 net_torque_new = calc_net_torque(k.vel, k.ang_vel);
+    Vector3 ang_accel_k;
+    ang_accel_k.x = net_torque_new.x / inertia[0];  // Separated bc inertia is an array, not a vector
+    ang_accel_k.y = net_torque_new.y / inertia[4];
+    ang_accel_k.z = net_torque_new.z / inertia[8];
+
+    rocket_.set_q_ornt(orient_true);  // resets the orientation to the initial
+
+    return {vel_k, accel_k, ang_vel_k, ang_accel_k};  // sets the values of the State structure
+}
+
+Quaternion<double> RungeKutta::calc_orient(double tStep, Vector3 ang_vel, Quaternion<double> orient) {
+    // Create a quaternion from angular velocity
+    Quaternion<double> q_omega{0, ang_vel.x, ang_vel.y, ang_vel.z};
+
+    // quaternion first time derivative
+    Quaternion<double> q_dot = 0.5 * q_omega * orient;
+
+    // Update orientation quaternion
+    orient = orient + (q_dot * tStep);
+
+    // Orientation quaternions must always stay at unit norm
+    orient.Normalize();
+
+    return orient;
+}
```

Comment:
Please copy my implementation of this exactly from [[AV-346]](https://github.com/ISSUIUC/ISS_SILSIM/pull/19) for consistency including the docstring.